### PR TITLE
Use HTTPS by default for connections to matomo.org

### LINKED ISSUE #19081: Use HTTPS by default for connections to matomo.org (with working fallback to HTTP requiring super user edit the INI config)
# Remaining steps

- Make HTTPS the default 
- Make sure the error message links to this FAQ: https://matomo.org/faq/how-to/faq-how-to-disable-https-for-matomo-org-and-api-matomo-org-requests/ 
- Update the FAQ https://matomo.org/faq/how-to/faq-how-to-disable-https-for-matomo-org-and-api-matomo-org-requests/ to mention it was put in 5.0.0 instead of "We will soon switch this connection over to HTTPS"
- Test that an update from 5.0.0-bX to 5.0.0-bY still work when HTTPS doesn't work (ie. people see the message, and after changing the INI config as per the clear FAQ, then it works for people)

# Initial issue

This applies to all api.matomo.org and plugins.matomo.org calls.

1. DONE  First we add a new required system check showing to users if the connection over HTTPS works or not for these endpoints. If it doesn't work, then there should be an error shown explaining that we will soon switch to HTTPS by default. They should either make HTTPS work or disable HTTPS (see next item). We should mention the consequences of not fixing this issue (eventually won't receive any updates anymore big security issue for sure, and using HTTP is a minor security issue that someone could pretend there is no longer an update available)

2. We introduce a setting to force HTTP instead of HTTPS as some people won't be able to change their PHP either because the hoster doesn't allow it or because they aren't technical enough etc.

3. Create an FAQ about how to make HTTPS work or disable HTTPS and link to it in the system check error message in 1 above.

### PR DESCRIPTION
**Pull request by bx80**
### Description:

Fixes #19081

- Global ini `api_service_url` default value updated to `https`
- Marketplace plugins.matomo.org default updated to `https`
- [FAQ](https://matomo.org/faq/how-to/faq-how-to-disable-https-for-matomo-org-and-api-matomo-org-requests/) updated
- Diagnostic warning and error messages link to the [FAQ](https://matomo.org/faq/how-to/faq-how-to-disable-https-for-matomo-org-and-api-matomo-org-requests/) 
- Config option `force_matomo_http_request = 1` will downgrade both the `api.matomo.org` and `plugins.matomo.org` protocol to `http` if set.

If for some reason a user has overridden the `api_service_url` config option in their `config.ini.php` and that url uses `http` then they will continue to use `http` and not default to `https` for requests to the matomo.org  api. It seems unlikely that this setting would be overridden without good reason.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/Installation/lang/en.json [hunk 1]
```diff
@@ -158,6 +158,7 @@
...
         "PerformanceSettingsDesc1": "Your Matomo is set up and ready to track and report on your website's traffic. Set up %1$sCLI archiving%2$s if you find it to be slow. This generates reports in the background, rather than on demand.",
         "PerformanceSettingsDesc2": "This requires adding a Matomo command to Cron which can't be done automatically by the installer. %1$sRead our FAQ to learn to set it up yourself.%2$s",
-        "MatomoHttpRequestConfigInfo": "The option force_matomo_http_request in the general config is turned on, we highly recommend switching it off for security reasons. For more details please read our %1$sFAQ%2$s"
+        "MatomoHttpRequestConfigInfo": "The option <code>force_matomo_http_request</code> in the general config is turned on, we highly recommend switching it off for security reasons. For more details please read %1$show to disable https for matomo.org and api.matomo org.requests%2$s",
```
Comment (sgiehl) [inline plugins/Installation/lang/en.json]:
```suggestion
        "MatomoHttpRequestConfigInfo": "The option <code>force_matomo_http_request</code> in the general config is turned on, we highly recommend switching it off for security reasons. For more details please read %1$show to disable https for matomo.org and api.matomo.org requests%2$s",
```

File: plugins/Installation/lang/en.json [hunk 2]
```diff
@@ -158,6 +158,7 @@
...
+        "MatomoHttpsNotSupported": "Could not initialize HTTPS. Connections to matomo.org use HTTPS by default for security. Please make sure that HTTPS works on your environment or enable the option `force_matomo_http_request`. For more details please read %1$show to disable https for matomo.org and api.matomo.org requests%2$s"
```
Comment (sgiehl) [inline plugins/Installation/lang/en.json]: Actually the config setting forces http connections to plugins.matomo.org as well. Maybe we should simply write `how to disable https for requests to matomo.org`

File: plugins/CoreUpdater/Diagnostic/HttpsUpdateCheck.php [hunk 3]
```diff
@@ -31,23 +31,26 @@ public function __construct(Translator $translator)
...
-        } else {
-            // successful using https
-            return array(DiagnosticResult::singleResult($label, DiagnosticResult::STATUS_OK));
+            // If the config option to force http is enabled then show info that it isn't recommended
+            $comment = $this->translator->translate('Installation_MatomoHttpRequestConfigInfo', $faqLink);
+            return [DiagnosticResult::singleResult($label, DiagnosticResult::STATUS_INFORMATIONAL, $comment)];
         }
+
+        if (!Http::isUpdatingOverHttps()) {
+            // Failed to request over https, show error
```
Comment (sgiehl) [inline plugins/CoreUpdater/Diagnostic/HttpsUpdateCheck.php]: That comment actually doesn't reflect what `Http::isUpdatingOverHttps` is doing. So either we change it or simply remove it.

**Review by sgiehl — CHANGES_REQUESTED** [review]
Left some minor comments. Otherwise this looks fine.
The issue also mentions to check if updating still works when https is not available. But guess that's something we can check if we have released a some first beta versions of Matomo 5

File: plugins/Installation/lang/en.json (hunk 2 above)
Comment (bx80) [reply plugins/Installation/lang/en.json]: Agreed, the various matomo.org subdomains aren't relevant to users. I've also adjusted the FAQ to match this (but not the FAQ url as the it's referenced in the previous release)

File: plugins/CoreUpdater/Diagnostic/HttpsUpdateCheck.php (hunk 3 above)
Comment (bx80) [reply plugins/CoreUpdater/Diagnostic/HttpsUpdateCheck.php]: Fixed :+1:

**Review by sgiehl — APPROVED** [review]

**Pull request merged**